src/App/Pages/Accounts/LoginPasswordlessViewModel.cs

```diff
@@ -63,8 +70,30 @@ public LoginPasswordlessDetails LoginRequest
             }
         }
 
+        public void StopRequestTimeUpdater() => _requestTimeCts?.Cancel();
+
+        private void StartRequestTimeUpdater()
+        {
+            _requestTimeCts?.Cancel();
+            _requestTimeCts = new CancellationTokenSource();
+            _requestTimeTask = new TimerTask(_logger, UpdateRequestTime, _requestTimeCts).RunPeriodic(TimeSpan.FromMinutes(5));
+        }
+
+        private async void UpdateRequestTime()
+        {
+            TriggerPropertyChanged(nameof(TimeOfRequestText));
+            if (DateTime.UtcNow > LoginRequest?.RequestDate.ToUniversalTime().AddMinutes(15))
+            {
+                StopRequestTimeUpdater();
+                await _platformUtilsService.ShowDialogAsync(AppResources.LoginRequestHasAlreadyExpired);
+                await Page.Navigation.PopModalAsync();
+                return;
+            }
+        }
+
         private async Task PasswordlessLoginAsync(bool approveRequest)
         {
+            StopRequestTimeUpdater();
```

Comment:
If this method fails throwing an exception at some point shouldn't the timer start again?

Reply:
Moved to the end of the method so that problem doesn't happen.